packages/ui/src/proposals/modals/AddNewProposal/components/StakingAccountStep.tsx

```diff
@@ -33,7 +34,11 @@ export const StakingAccountStep = ({ requiredStake, account: chosenAccount, setA
             </TextMedium>
           </RowGapBlock>
           <InputComponent label="Select account for Staking" required inputSize="l">
-            <SelectAccount onChange={(account) => setAccount(account)} selected={chosenAccount} />
+            <SelectAccount
+              onChange={(account) => setAccount(account)}
+              selected={chosenAccount}
+              filter={(account) => useBalance(account.address)?.transferable.gte(requiredStake) ?? true}
```

Comment:
Two things here:

1.  Hooks must be placed at the top of the component since [rules of the hooks](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) (there's a slight possibility that hooks will render in a different order).
2.  Please extract this check – because it might be not that simple. For instance, we should check (later) if there is a lock that would allow reusing locked balance for this purpose.

Reply:
I've decided to introduce new prop `minBalance` for selected account component, I think it should be more clear than writing filter each time